src/deepagents/middleware/filesystem.py

```diff
@@ -568,54 +567,69 @@ async def awrap_model_call(
             request.system_prompt = request.system_prompt + "\n\n" + self.system_prompt if request.system_prompt else self.system_prompt
         return await handler(request)
 
-    def _intercept_large_tool_result(self, tool_result: ToolMessage | Command) -> ToolMessage | Command:
+    def _process_large_message(
+        self,
+        message: ToolMessage,
+        resolved_backend: BackendProtocol,
+    ) -> tuple[ToolMessage, dict[str, FileData] | None]:
+        content = message.content
+        if not isinstance(content, str) or len(content) <= 4 * self.tool_token_limit_before_evict:
+            return message, None
+
+        file_path = f"/large_tool_results/{message.tool_call_id}"
```

Comment:
could you sanitize the tool call id by replacing catching `..` and `/` and `\` and replacing them with something less threatening?

Reply:
Also how do we handle separators?